src/main/java/net/openhft/chronicle/wire/LongConverter.java

```diff
@@ -141,6 +155,19 @@ default void lengthCheck(CharSequence text) {
             throw new IllegalArgumentException(format("text={0} exceeds the maximum allowable length of {1}", text, maxParseLength()));
     }
 
+    /**
+     * Checks that the length of the provided text does not exceed the allowable maximum.
+     *
+     * @param beginIndex the beginning index, inclusive.
+     * @param endIndex the ending index, exclusive.
+     * @throws IllegalArgumentException if the text length exceeds the maximum allowable length.
+     */
+    default void lengthCheck(int beginIndex, int endIndex) {
```

Comment:
Nice! Please add some explicit test coverage for the bounds checking if we don't have it already